src/calliope/backend/helper_functions.py

```diff
@@ -31,7 +30,7 @@ def __init__(
         *,
         equation_name: str,
         input_data: xr.Dataset,
-        backend_interface: Optional[type[BackendModel]] = None,
+        backend_interface=None,
```

Comment:
For some reason this ignores `TYPE_CHECKING`. So I removed it.

Reply:
Could you try `Optional[type["BackendModel"]]`?